src/types.ts

```diff
@@ -12,8 +14,9 @@ export interface SmartweaveContractSource {
 }
 
 export type SearchBarProps = {
-  buttonAction: () => void;
+  buttonAction: (props: string) => void;
   placeholderText: string;
-  headerText: string;
+  headerElement: JSX.Element;
```

Comment:
we could even got more explicit here and make it have to be `SearchBarHeader` component